File: 2024-04-04-build-rag-with-python/import.py
import ollama, chromadb, time
from utilities import readtext, getconfig
from mattsollamatools import chunker, chunk_text_by_sentences


chroma = chromadb.HttpClient(host="localhost", port=8000)
chroma.delete_collection("buildragwithpython")
collection = chroma.get_or_create_collection(name="buildragwithpython", metadata={"hnsw:space": "cosine"})

embedmodel = getconfig()["embedmodel"]
starttime = time.time()
with open('sourcedocs.txt') as f:
  lines = f.readlines()
  for filename in lines:
    text = readtext(filename)
    chunks = chunk_text_by_sentences(source_text=text, sentences_per_chunk=7, overlap=0 )
    print(f"with {len(chunks)} chunks")
    for index, chunk in enumerate(chunks):
      embed = ollama.embeddings(model=embedmodel, prompt=chunk)['embedding']
      print(".", end="", flush=True)
      collection.add([filename+str(index)], [embed], documents=[chunk], metadatas={"source": filename})
    
print("--- %s seconds ---" % (time.time() - starttime))

File: 2024-04-08-build-rag-with-typescript/import.ts
import ollama from "ollama";
import { ChromaClient } from "chromadb";
import { getConfig, readText } from "./utilities";
import { chunkTextBySentences } from "matts-llm-tools";

const chroma = new ChromaClient({ path: "localhost:8000" });
await chroma.deleteCollection({ name: "buildragwithtypescript" });
const collection = await chroma.getOrCreateCollection({ name: "buildragwithtypescript", metadata: { "hnsw:space": "cosine" } });

const { embedmodel, mainmodel } = getConfig();

const docstoimport = (await Bun.file("sourcedocs.txt").text()).split("\n");
for (const doc of docstoimport) {
  console.log(`\nEmbedding chunks from ${doc}\n`)
  const text = await readText(doc)
  const chunks = chunkTextBySentences(text, 7, 0);


  for await (const [index, chunk] of chunks.entries()) {
    const embed = (await ollama.embeddings({ model: embedmodel, prompt: chunk })).embedding
    await collection.add({ ids: [doc + index], embeddings: [embed], metadatas: { source: doc }, documents: [chunk] })
    process.stdout.write(".")
  }
}
